### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by jordan-a-young**
Majority of the updates are for devDependencies like `tsup`, `jest`, and `nx`. Any dependencies that were updated were only minor or patch updates. Therefore I didn't mark any of the version updates as breaking.

GitHub Actions now run for node 16 and 18 with 18 as the version for publish

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "lint:fix": "nx affected --target=lint --fix",
     "new": "plop",
     "nx": "nx",
-    "postinstall": "is-ci || husky install",
+    "prepare": "is-ci || husky install",
```
Comment (gregmartDOTin) [inline package.json:41]: Did this change again? I thought on yarn 2+ we were supposed to use postinstall: https://typicode.github.io/husky/getting-started.html#yarn-2

File: packages/analytics-core/package.json [hunk 2]
```diff
@@ -29,12 +29,12 @@
     "publish": "yarn npm publish --tolerate-republish --access public"
   },
   "dependencies": {
-    "yup": "^0.32.9"
+    "yup": "^0.32.11"
```
Comment (gregmartDOTin) [inline packages/analytics-core/package.json:32]: Can we use yup 1.x?

File: package.json (hunk 1 above)
Comment (jordan-a-young) [reply package.json:41]: I missed that note in the docs. I'll switch it back

File: packages/analytics-core/package.json (hunk 2 above)
Comment (jordan-a-young) [reply packages/analytics-core/package.json:32]: I think we should do it in its own ticket. I've had too many headaches with `yup` in the past

**Review by gregmartDOTin — APPROVED** [review]

**Pull request merged**